test/xUnit/csharp/test_PowerShell.cs

```diff
@@ -0,0 +1,167 @@
+
+namespace PSTests.Parallel
+{
+    public class PowerShellTests
+    {
+        [Fact]
+        public static async Task TestPowerShellInvokeAsync()
+        {
+            using (var ps = PowerShell.Create())
+            {
+                ps.AddCommand("Get-Process")
+                  .AddParameter("Id", Process.GetCurrentProcess().Id);
+
+                var results = await ps.InvokeAsync();
+
+                Assert.Single(results);
+                Assert.IsType<Process>(results[0]?.BaseObject);
+                Assert.Equal(Process.GetCurrentProcess().Id, ((Process)results[0].BaseObject).Id);
+            }
+        }
+
+        [Fact]
+        public static async Task TestPowerShellInvokeAsyncWithInput()
+        {
+            using (var ps = PowerShell.Create())
+            {
+                ps.AddCommand("Get-Command");
+
+                var results = await ps.InvokeAsync(new PSDataCollection<string>(new[] { "Get-Process" }));
+
+                Assert.Single(results);
+                Assert.IsType<CmdletInfo>(results[0]?.BaseObject);
+                Assert.Equal("Get-Process", ((CmdletInfo)results[0].BaseObject).Name);
+            }
+        }
+
+        [Fact]
+        public static async Task TestPowerShellInvokeAsyncWithInputAndOutput()
+        {
+            using (var ps = PowerShell.Create())
+            {
+                ps.AddCommand("Get-Command");
+
+                var results = new PSDataCollection<CmdletInfo>();
+                await ps.InvokeAsync(new PSDataCollection<string>(new[] { "Get-Process" }), results);
+
+                Assert.Single(results);
+                Assert.IsType<CmdletInfo>(results[0]);
+                Assert.Equal("Get-Process", results[0].Name);
+            }
+        }
+
+        [Fact]
+        public static async Task TestPowerShellInvokeAsyncWithErrorInScript()
+        {
+            using (var ps = PowerShell.Create())
+            {
+                ps.AddScript(@"
+try {
+    Get-Process -Invalid 42 -ErrorAction Stop
+} catch {
+    throw
+}
+");
+                var results = await ps.InvokeAsync();
+
+                Assert.True(results.IsFaulted);
+                Assert.IsType<AggregateException>(results.Exception);
+                Assert.IsType<ParameterBindingException>(results.Exception.InnerException);
+                Assert.Equal(results.Exception.InnerException.CommandInvocation.InvocationName, "Get-Process");
+                Assert.Equal(results.Exception.InnerException.ParameterName, "Invalid");
+                Assert.Equal(results.Exception.InnerException.ErrorId, "NamedParameterNotFound");
+            }
+        }
+
+        [Fact]
+        public static async Task TestPowerShellInvokeAsyncInUnopenedRunspace()
+        {
+            using (var rs = RunspaceFactory.CreateRunspace())
+            using (var ps = PowerShell.Create(rs))
+            {
+                var results = await ps.AddScript(@"@(1..10).foreach{Start-Sleep -Milliseconds 500}")
```

Comment:
Please change this test script to something simple, like just `1+1`. The test script here doesn't really matter since the invoke will fail right away.

Reply:
Same comment as previous (waiting for confirmation on necessity for C# tests).